impeller/renderer/backend/vulkan/swapchain_impl_vk.cc

```diff
@@ -377,8 +378,32 @@ bool SwapchainImplVK::Present(const std::shared_ptr<SwapchainImageVK>& image,
   }
 
   const auto& context = ContextVK::Cast(*context_strong);
+  auto current_frame = current_frame_;
+  const auto& sync = synchronizers_[current_frame];
 
-  const auto& sync = synchronizers_[current_frame_];
+  // Submit all command buffers.
+  {
+    auto [fence_result, fence] = context.GetDevice().createFenceUnique({});
+    if (fence_result != vk::Result::eSuccess) {
+      return false;
+    }
+
+    const auto& encoders = context.GetCommandBufferQueue()->Take();
```

Comment:
Related to my question above, i guess if this is guaranteed to be called before encoding another image, it should be fine for the context to have one command buffer queue.  I don't know if that's the case off the top of my head.